### PR TITLE
Add support upload collections

### PR DESCRIPTION
**Pull request by pingsutw**
# TL;DR
Add support upload collection (a list).
When using raw container in the map task, flytepropeller expect the [subtask upload a list](https://github.com/flyteorg/flyteplugins/blob/f577f62727a16a282edfa7a49d985e47a4ccef79/go/tasks/plugins/array/outputs.go#L121).

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
^^^

## Tracking Issue
https://flyte-org.slack.com/archives/CP2HDHKE1/p1678230956906899

## Follow-up issue
_NA_

### CODE REVIEW

**Comment by wild-endeavor** [comment]
before merging, let's copy the minimum copilot and propeller versions into the PR description.

**Review by wild-endeavor — APPROVED** [review]
+1 but a bit confused... why do we need to do this?  I thought for map tasks propeller is the one that assembles the list. and the individual tasks still upload single scalar items.

**Comment by pingsutw** [comment]
propeller expects output is a list with only one item. https://github.com/flyteorg/flyteplugins/blob/f577f62727a16a282edfa7a49d985e47a4ccef79/go/tasks/plugins/array/outputs.go#L121

**Comment by pingsutw** [comment]
because flytekit will convert output to a list literal. https://github.com/flyteorg/flytekit/blob/93995e29dcd37317e91ebcfb3019a4ffed0e82de/flytekit/core/map_task.py#LL56C47-L56C47
